src/pathogen_embed/pathogen_embed.py

```diff
@@ -371,13 +367,15 @@ def cluster(args):
         plot_data["cluster"] = clusterer.labels_.astype(str)
 
         plot_df = pd.DataFrame(plot_data)
-        ax = sns.scatterplot(
-            data=plot_df,
-            x="x",
-            y="y",
-            hue="cluster",
-            alpha=0.5,
-        )
+        clusters = plot_df['cluster'].unique()
+        colors = plt.cm.tab10.colors[:len(clusters)]
+        for i, cluster in enumerate(clusters):
+            cluster_data = plot_df[plot_df['cluster'] == cluster]
+            plt.scatter(cluster_data["x"], cluster_data["y"], color=colors[i], label=f'Cluster {cluster}', alpha=0.5)
```

Comment:
I just pushed this change, since I had to make it locally to confirm that it actually worked anyway. I'll merge the PR now and we can discuss the numba version separately.